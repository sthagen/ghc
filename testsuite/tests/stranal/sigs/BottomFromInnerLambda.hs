module BottomFromInnerLambda where

expensive :: Int -> Int
expensive 0 = 0
expensive n = expensive n
{-# NOINLINE expensive #-}

-- We could be saying "<1!P(SA)><A>b"
-- but we are saying "<1!P(SA)>"
-- We should not be saying "<1!P(SA)>b"
f :: Int -> Int -> Int
f x = expensive x `seq` (\y -> error (show y))
