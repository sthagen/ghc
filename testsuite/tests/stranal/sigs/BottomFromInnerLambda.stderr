
==================== Strictness signatures ====================
BottomFromInnerLambda.$trModule:
BottomFromInnerLambda.expensive: <1!P(SA)>
BottomFromInnerLambda.f: <1!P(SA)>



==================== Cpr signatures ====================
BottomFromInnerLambda.$trModule:
BottomFromInnerLambda.expensive: 1
BottomFromInnerLambda.f:



==================== Strictness signatures ====================
BottomFromInnerLambda.$trModule:
BottomFromInnerLambda.expensive: <1!P(1A)>
BottomFromInnerLambda.f: <1!P(1A)>


